CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -732,7 +731,7 @@ public static function formRule($fields, $files, $self) {
                   if (array_key_exists('membership_type_id', $fieldValue['options'][$key])
                     && in_array($fieldValue['options'][$key]['membership_type_id'], $currentMemberships)
                   ) {
-                    $errors['price_' . $fieldKey] = ts($errorText, [1 => CRM_Member_PseudoConstant::membershipType($fieldValue['options'][$key]['membership_type_id'])]);
+                    $errors['price_' . $fieldKey] = ts('Your %1 membership was previously cancelled and can not be renewed online. Please contact the site administrator for assistance.', [1 => CRM_Member_PseudoConstant::membershipType($fieldValue['options'][$key]['membership_type_id'])]);
```

Comment:
I understand that the author meant to reduce code duplication, but in this case, it meant that the string was not extracted for translation.

Might be worth creating a function for these conditions, so that it could be something along the lines of "if (wasPreviouslyCancelled()) { $errors[..] = ts('the message');".